src/main/java/com/alysaa/geyseradmintools/listeners/AdminToolInventory.java

```diff
@@ -12,48 +11,39 @@
 import org.bukkit.event.inventory.InventoryClickEvent;
 import org.bukkit.event.player.PlayerDropItemEvent;
 import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.ItemStack;
 
 
 public class AdminToolInventory  implements Listener {
 
+    private static final FileConfiguration config = Gat.plugin.getConfig();
+    private static final ItemStack starTool = ItemStackFactory.getStarTool();
+
     @EventHandler
     public void onInventoryClick(InventoryClickEvent e) {
-        for (Player player : Bukkit.getOnlinePlayers()) {
-            if (!player.hasPermission("geyseradmintools.item")) {
-                return;
-            }
-            FileConfiguration config = Gat.plugin.getConfig();
-            if (!config.getBoolean("DisableItemMove")) {
-                return;
+        if (config.getBoolean("DisableItemMove")) {
+            if (e.getCurrentItem().equals(starTool)) {
+                e.setCancelled(true);
             }
```

Comment:
I'll switch these 2 if statements... Its probably more common for the second statement to fail than the first.